src/hotspot/share/opto/vectornode.cpp

```diff
@@ -248,14 +248,6 @@ int VectorNode::opcode(int sopc, BasicType bt) {
     return Op_StoreVector;
   case Op_MulAddS2I:
     return Op_MulAddVS2VI;
-  case Op_ConvI2F:
-    return Op_VectorCastI2X;
-  case Op_ConvL2D:
-    return Op_VectorCastL2X;
-  case Op_ConvF2I:
-    return Op_VectorCastF2X;
-  case Op_ConvD2L:
-    return Op_VectorCastD2X;
```

Comment:
@vnkozlov @sviswa7 , I removed these lines here, because I call another api to generate vector opcodes and vector nodes for conversion nodes [VectorCastNode::opcode()](https://github.com/openjdk/jdk/blob/ba9ee8cb286268f1d6a2820508334aaaf3131e15/src/hotspot/share/opto/vectornode.cpp#L1203), which is more precise than this one and unified with Vector API. Because converting short to double and converting int to double would use the same opcode `ConvI2D` for scalar nodes, we can't differentiate between them just based on the opcode of the current node. I adjusted the callers `VectorCastNode::implemented()` and `SuperWord::output()` to the removing.

Reply:
> May be we should have assert here to make sure that in all places we call `VectorCastNode::opcode()` for `Conv*` nodes:
> 
> ```
>  default:
>     assert(!VectorNode::is_convert_opcode(sopc), "Convert node %s should be processed by VectorCastNode::opcode()", NodeClassNames[sopc]);
>     return 0; // Unimplemented
> ```

Make sense to me. Thanks!